docs/models/overview.md

```
@@ -460,6 +462,53 @@ fallback_model = FallbackModel(
+    fallback_on=fall_back_unless_primary_auth_error,
+)
+agent = Agent(fallback_model)
+```
+
+The second argument is optional and detected per handler, so existing single-argument handlers keep working unchanged.
+
+### Attributing Failures to Models
+
+When all models fail, the resulting [`FallbackExceptionGroup`][pydantic_ai.exceptions.FallbackExceptionGroup] wraps each model's exception in order of attempt, matching the order of the models passed to `FallbackModel`. Each sub-exception is annotated with a [PEP 678](https://peps.python.org/pep-0678/) note identifying the model that raised it, so tracebacks (and `exception.__notes__`) tell you exactly which model produced which error:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)